### PR TITLE
fix: opportunities selectors sorting

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR fixes the wrong sorting of LP/staking opportunities in the DeFi opportunities selectors, which never properly worked and was made obvious by UNI-V2 LP opportunities.

tl;dr is that lodash `sortBy()` doesn't handle floats nicely, and we were effectively getting the sorting totally wrong.

Fixed by getting the active (positive fiat amount chunks) separately, and the inactive (0 amount) separately, using native JS and BN methods for the comparison. Inactive chunks which are then `Array.prototype.sort`ed once again by net APY

While at it:
- adds sorting of LP opportunities. While the aggregated staking selector we're consuming is sorting them, the LP one was doing no sorting at all, resulting in a random order for LP opportunities which was made obvious by the addition of UNI-V2 opportunities
- removes the default react-table sorting, which sorts by a single criteria only, and isn't able to handle the "sort by fiat amount descending, then sort inactives (0 `fiatAmount` by APY) sorting we need. Again, this seemed like it was working previously only because of the fewer amount of opportunities we had, this really was missing the double-sorting altogether

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Low, this might bork sorting by APY, test accordingly

## Testing

- DeFi opportunities by provider are sorted correctly
- DeFi opportunities by asset are sorted correctly
- DeFi providers are sorted by the highest total value over its opportunities
- Defi assets are sorted by the highest total value over its opportunities 

## Screenshots (if applicable)

image
image
image

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4397** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4397" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4397?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
WIPing this guy while I context switch to CRs, fixed the guts of it, but it looks as `searchOpportunities()` is now rugging the sorting

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts [hunk 1]
```diff
@@ -541,11 +540,19 @@ export const selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty =
...
+
+      const sortedResultsByFiatAmount = results.sort((a, b) =>
+        bnOrZero(a.fiatAmount).gte(bnOrZero(b.fiatAmount)) ? -1 : 1,
+      )
+
+      const activeResults = sortedResultsByFiatAmount.filter(opportunity =>
+        bnOrZero(opportunity.fiatAmount).gt(0),
+      )
+      const inactiveResults = sortedResultsByFiatAmount
+        .filter(opportunity => bnOrZero(opportunity.fiatAmount).eq(0))
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:552]: for all of these active vs inactive we can partition if we want https://lodash.com/docs/4.17.15#partition

File: src/state/slices/txHistorySlice/txHistorySlice.ts
Comment (0xdef1cafe) [inline src/state/slices/txHistorySlice/txHistorySlice.ts:1]: we may want to make this change in separate PR from the sorting, and probably don't need to `Promise.all` at all here

**Review by 0xdef1cafe — APPROVED** [review]
conditional approval if you want to revert foxy changes for a separate PR as it's somewhat unrelated to sorting here

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts (hunk 1 above)
Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts]: 6c33866aa2b84f4fbd026cdcb8a0f880b7a2c076

File: src/state/slices/txHistorySlice/txHistorySlice.ts
Comment (gomesalexandre) [reply src/state/slices/txHistorySlice/txHistorySlice.ts:1]: Reverted in 9728126a9a832fc878fe04543eb7011c331b7ba2 and will tackle both in a separate follow-up PR - tested gomeyeet and confirmed perf is still happy here

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by 0xApotheosis** [comment]
@gomesalexandre I don't have any hard data on it, but I suspect this PR is causing performance issues.

Of note:

Screenshot 2023-05-05 at 12 05 22 pm
